lms/services/grouping.py

```diff
@@ -33,53 +33,43 @@ def generate_authority_provided_id(
             tool_consumer_instance_guid, parent.lms_id, type_.value, lms_id
         )
 
-    def upsert_with_parent(  # pylint: disable=too-many-arguments
-        self,
-        tool_consumer_instance_guid,
-        lms_id,
-        lms_name,
-        parent: Grouping,
-        type_: Grouping.Type,
-        extra=None,
-    ):
+    def upsert_with_parent(self, grouping_dicts: List[dict]):
         """
         Upsert a Grouping generating the authority_provided_id based on its parent.
 
-        :param tool_consumer_instance_guid: Tool consumer GUID
```

Comment:
No longer taking `tool_consumer_instance_guid`. Using the one from `self.application_instance`.

Reply:
I wonder about the consequences of this change

Reply:
I think it should only have consequences if we where doing the wrong thing before, if `application_instance.tool_consumer_instance_guid != tool_consumer_instance_guid`.

Can't tell if that has happen in the past. I doubt it as you will have to come up with a tool_consumer_instance_guid from somewhere. But of course we've seen lots of weird edge cases.